src/components/filter/EventsFilterPanel.tsx

```diff
@@ -45,60 +52,81 @@ function EventsFilterPanel() {
 	const eventsStore = useWorkspaceEventStore();
 	const eventDataStore = useEventsDataStore();
 	const filterStore = useEventsFilterStore();
+	const { addHistoryItem } = useFiltersHistoryStore();
+	const { autocompletes, addFilter } = useEventFilterAutocompletesStore();
 
 	const [showFilter, setShowFilter] = React.useState(false);
 
-	const [filter, setFilter] = React.useState<EventFilterState | null>(filterStore.filter);
 	const [currentFilterValues, setCurrentFilterValues] = React.useState<CurrentFilterValues | null>(
 		getDefaultCurrentFilterValues(filterStore.filter),
 	);
 
 	React.useEffect(() => {
-		setFilter(filterStore.filter);
 		setCurrentFilterValues(getDefaultCurrentFilterValues(filterStore.filter));
 	}, [filterStore.filter]);
 
 	const onSubmit = React.useCallback(() => {
-		if (filter) {
-			eventsStore.applyFilter(filter);
+		if (filterStore.temporaryFilter) {
+			const { status, ...restFilters } = filterStore.temporaryFilter;
+			const filtersToSave = Object.fromEntries(
```

Comment:
This saving applied filter logic has nothing to do with this component, you have a store that should handle this